appium.js

```diff
@@ -73,12 +91,91 @@ function connectGetDefaultArgs () {
   });
 }
 
+function connectStartSession (win) {
+  ipcMain.on('start-session', () => {
+    let sessionWin = new BrowserWindow({width: 800, height: 600, webPreferences: {devTools: true}});
+    let sessionHTMLPath = path.resolve(__dirname, 'app', 'index.html#/session');
+    sessionWin.loadURL(`file://${sessionHTMLPath}`);
+    sessionWin.show();
+
+    // When you close the session window, kill the associated Appium session
+    let sessionWinID = sessionWin.webContents.id;
+    sessionWin.on('closed', () => {
+      killSession(sessionWinID);
```

Comment:
The way that a session is killed is by closing the window. 'appium.js' subscribes to the window closing event and then calls 'killSession' so the window will be gone before they can see a loading indicator.

Also, you can't use UI indicators in 'appium.js' because it runs on the main process.